tests/set-namespace/no-fnconfig-resource/.expected/results.yaml

```diff
@@ -0,0 +1,14 @@
+apiVersion: kpt.dev/v1
+kind: FunctionResultList
+metadata:
+  name: fnresults
+exitCode: 0
+items:
+  - image: gcr.io/kpt-fn/set-namespace:unstable
+    exitCode: 0
+    results:
+      - message: FunctionConfig is missing. Expect `ConfigMap.v1` or `SetNamespace.v1alpha1.fn.kpt.dev`
+        severity: error
+        resourceRef: {}
+        file:
+          index: -1
```

Comment:
Hmm... What does `-1` mean?

Reply:
This is intended. file index -1 meaning the file path does not exist.  This test covers the cases where functionConfig is required but not given

Reply:
Quoted from the functions spec: https://github.com/kubernetes-sigs/kustomize/blob/master/cmd/config/docs/api-conventions/functions-spec.md

> index:
            type: number
            default: 0
            description: Index of the object in a multi-object YAML file.

The function spec has not define what does `-1` mean. `kpt` has not either.

> functionConfig is required but not given

IMO it's too hacky to use `-1` to communicate this. We can leave the `index` field empty and use the `message` field to communicate with the user.

Reply:
I don't think using `-1` is the right thing to do here. But I'm OK if you want to fix this in a followup PR.

Reply:
Do you know what index value kpt uses to mark resources from generator?